### ISSUE TITLE
Basic authorization is not allowed while webAuthn enrolled

### ISSUE THREAD

**Issue by markburgessstl** [go-gitea/gitea#32895]
### Description

I created a passkey to login to the web ui.  Once I do that, I can successfully login.  However, I can bo longer access my repos using the Git command line, see below:

Username for 'https://mini.mbtest.local': markb
Password for 'https://markb@mini.mbtest.local': 
remote: Verify
fatal: Authentication failed for 'https://mini.mbtest.local/gogs/markb/buffy_mstdn.git/'

In the Gitea log I am seeing:

2024/12/18 13:48:26 ...eb/routing/logger.go:102:func1() [I] router: completed GET /markb/buffy_mstdn.git/info/refs?service=git-upload-pack for 192.168.0.2:60311, 401 Unauthorized in 4.7ms @ repo/githttp.go:518(repo.GetInfoRefs)
2024/12/18 13:48:30 routers/web/web.go:120:func9() [E] Failed to verify user: Basic authorization is not allowed while webAuthn enrolled
2024/12/18 13:48:30 ...eb/routing/logger.go:102:func1() [I] router: completed GET /markb/buffy_mstdn.git/info/refs?service=git-upload-pack for 192.168.0.2:60311, 401 Unauthorized in 48.8ms @ web/web.go:117(web.Routes.webAuth)
2024/12/18 13:48:30 routers/web/web.go:120:func9() [E] Failed to verify user: Basic authorization is not allowed while webAuthn enrolled

When I delete the webauthn security key from my user settings, I can then access via git with no problem.

Bottom line if we can't use basic auth then at the very least you should provide a warning.

### Gitea Version

 1.23.0-rc0

### Can you reproduce the bug on the Gitea demo site?

No

### Git Version

2.47.1

### Operating System

MacOS 15.2

### How are you running Gitea?

Downloaded the MacOS ARM version.  Running behind Apache

### Database

MySQL/MariaDB

**Comment by markburgessstl** [comment]
Also, I think you should be able to use passkeys just for accessing the GUI and other Git supported authentication methods from the command line.  Totally different use cases.

**Comment by lunny** [comment]
Have you tried using your user name and an access token? When 2FA/PASSkey is enabled, password pull/push will not be enabled.

**Comment by markburgessstl** [comment]
> Have you tried using your user name and an access token? When 2FA/PASSkey is enabled, password pull/push will not be enabled.

That worked, thank you!  I had no idea.

**Comment by techknowlogick** [comment]
Yup. This is intentional as eventually(tm) passwords will be disabled for cloning. This prevents a whole class of ways that passwords leak. It also allows for scoping of the token. 
There are even some clients that will do oauth so it's a short-lived access token.
Will definitely add this to the list of things to be documented. Thanks for bringing it up :)
